### PR TITLE
Allow Setting Detection Delay

### PR DESCRIPTION
**Pull request by DiegoxK**
This pull request addresses issue #12 and #10 by introducing a configurable detection delay for the extension. Users can now set a delay between a range to control the frequency at which the extension reacts to errors and voice lines are played.

I recommend a minimum cooldown duration of `600000` milliseconds (10 minutes) and a maximum of `3600000` (1 hour)  for really unexpected remainders of your failures **c:**

### CODE REVIEW

File: src/ErrorTracker.ts [hunk 1]
```diff
@@ -9,20 +11,43 @@ export function OnDidChangeDiagnostics(e: DiagnosticChangeEvent): void {
...
+    Math.random() * (maxDuration - minDuration + 1) + minDuration
+  );
+  return randomDuration;
+}
+
 function errorsForFile(editor: TextEditor, fileUri: Uri): Array<Diagnostic> {
 	return languages
 		.getDiagnostics(fileUri)
 		.filter(d => d.severity === DiagnosticSeverity.Error && isErrorCurrent(editor, d));
+
```
Comment (89netraM) [inline src/ErrorTracker.ts:50]: Stray newline

Comment (DiegoxK) [reply src/ErrorTracker.ts]: Should be removed now

**Comment by DiegoxK** [comment]
I have tested the changes, and setting both the maximum and minimum values to 0 produces the default behavior where the voices can be heard for each error

**Review by 89netraM — APPROVED** [review]
Looks good to me. I've got the same issue with running it on my PC, but if it sounds good to you then 👍

**Comment by DiegoxK** [comment]
Thank you! I believe I encountered an error with npm while compiling the speaker module, which was related to the installation of Visual Studio Code's C++ extension.

**Comment by 89netraM** [comment]
I'm terrible sorry, I feel so negative, but the `package-lock.json` file was lost in the merge.

**Comment by DiegoxK** [comment]
Yeah, I can see. Don't worry, you're not being negative at all. Let me check and upload the package.json uwu

**Comment by DiegoxK** [comment]
it should be uploaded now, I'm so sorry as well. I'm kinda new to collaborating to open source

**Comment by 89netraM** [comment]
No worries, all's good! I'm kinda new too.

**Comment by NinjaZMY** [comment]
@89netraM still not working for me, I'm using windows by the way
this might be due to the code becoming obsolete out-dated

**Pull request merged**